astropy/stats/circstats.py

```diff
@@ -0,0 +1,490 @@
+
+    if(np.size(alpha, axis=ax_alpha) != np.size(beta, axis=ax_beta)):
+        raise ValueError("alpha and beta must be arrays of the same size")
+
+    sin_a = np.sin(alpha - circmean(alpha, w_alpha, ax_alpha))
+    sin_b = np.sin(beta - circmean(beta, w_beta, ax_beta))
+    rho = np.sum(sin_a*sin_b)/np.sqrt(np.sum(sin_a*sin_a)*np.sum(sin_b*sin_b))
+
+    return rho
+
+def rayleightest(data, w=None, axis=None):
+    """ Performs the Rayleigh test of uniformity.
+
+    This test is  used to indentify a non-uniform distribution, i.e. it is
+    designed for detecting an unimodal deviation from uniformity. More
+    precisely, it assumes the following hypotheses:
+    - H0 (null hypothesis): The population is distributed uniformly around the
+    circle.
+    - H1 (alternative hypothesis): The population is not distributed uniformly
+    around the circle.
+    Small p-values suggest to reject the null hypothesis.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Array of circular (directional) data measured in radians.
+    w : numpy.ndarray, optional
+        In case of grouped data, the i-th element of ``w`` represents a
+        weighting factor for each group such that sum(w,axis) equals the number
+        of observations. See [1], remark 1.4, page 22, for detailed
+        explanation.
+    axis : int, optional
+        Axis along which the Rayleigh test will be performed.
+
+    Returns
+    -------
+    p-value : float
+        p-value.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from astropy.stats import rayleightest
+    >>> data = np.array([3.8955614, 3.1700932, 3.1804325, 2.7887885, 2.9513829,
+    ...                  3.1140079, 4.5052974, 3.7984484, 3.7510773, 2.9764646,
+    ...                  2.2100108, 2.9529469, 2.9722527, 3.5099678, 2.6844710,
+    ...                  4.5020762, 2.9285673, 2.2871037, 3.5805262, 3.4247286,
+    ...                  2.4355773, 3.7549614, 4.1346585, 2.8426607, 3.3291801,
+    ...                  2.5847791, 2.7217270, 3.4323084, 3.3058256, 3.2147845,
+    ...                  2.4918005, 3.4849414, 0.8249985, 3.4881397, 3.2070389,
+    ...                  3.0859854, 4.0566486, 2.3463984, 3.7370984, 5.6853310,
+    ...                  5.8108009, 3.3921987, 3.2166975, 3.6827617, 4.5055291,
+    ...                  3.5295258, 3.0162183, 3.2317242, 3.2778354, 3.1713455])
+    >>> rayleightest(data) # doctest: +FLOAT_CMP
+    2.726928722464598e-13
+
+    References
+    ----------
+    .. [1] S. R. Jammalamadaka, A. SenGupta. "Topics in Circular Statistics".
+       Series on Multivariate Analysis, Vol. 5, 2001.
+    .. [2] C. Agostinelli, U. Lund. "Circular Statistics from 'Topics in
+       Circular Statistics (2001)'". 2015.
+       <https://cran.r-project.org/web/packages/CircStats/CircStats.pdf>
+    .. [3] M. Chirstman., C. Miller. "Testing a Sample of Directions for
+       Uniformity." Lecture Notes, STA 6934/5805. University of Florida, 2007.
```

Comment:
Not that this isn't a useful reference as-is.  I was able to Google for it, and find it helpful.  But is there not a reference for this that is less ephemeral than some lecture notes, which have a habit of disappearing?  

Basically, if there are magic numbers appearing in some code, those numbers have been be well-substantiated in the code itself, or at least reference an easy to find book or something, because otherwise the code is at high risk of becoming a black box.  Maybe I'm over-thinking it in this case though.  I don't know how common this derivation is.